docs/blueprints/forgotten-deletes.md

```diff
@@ -0,0 +1,124 @@
+
+### Delete pieces requests
+
+Delete pieces requests are issued when a user deletes an object. The metainfo service deletes the object's pointers and then sends delete pieces requests to each of the reliable nodes which held pieces. The rest of the nodes' pieces will be eventually deleted by garbage collection. 
+
+To prove to the satellite that a piece was deleted legitimately, we will have it sign the delete pieces requests. The node will present these to the satellite in the event of an audit for a deleted piece.
+
+```
+message DeletePiecesRequest {
+    repeated bytes piece_ids = 1 [(gogoproto.customtype) = "PieceID", (gogoproto.nullable) = false];
+    bytes satellite_signature = 2;
+}
+```
+
+The storage nodes will serialize and save these signed messages into a file store.
+
+#### File store
+
+// TODO: directory names
+Within the storage node's 'storage' directory, we will add a new directory called 'deleted-pieces-proof'. Within this directory we will create a subdirectory for each satellite, and within each of these we will add two more directories: 'pieces' and 'signed-requests'.
+
+##### Creating Proofs
+
+The DeletePieces method on the piecestore endpoint is where the the delete pieces requests are sent. From here, the piece IDs within the message are enqueued for the piece deleter job to delete. Before the pieces are enqueued, we will write the serialized message into a file in the 'signed-requests' directory of the respective satellite, skipping the first byte which we will reserve. Then we pass this filepath along with the piece IDs into the Enqueue method. As pieces in the queue are deleted, we create a new path for them in the 'pieces' directory. At this path, we write a text file containing the name of the signed delete request file which contains this piece. Then, we open the signed delete request file and increment the first byte.
+
+##### Deleting Proofs
+
+We can use garbage collection to determine when to remove the deleted piece IDs and their signed proofs. Due to the aforementioned changes to how we run garbage collection, we can be sure that any piece which does not exist in the filter will not be reverted. Thus, once the filters no longer contain the deleted pieces, the piece ID paths can be removed. To do this, we will have the 'Retain' code path walk and clean up the deleted pieces directory just as it does the 'blobs' directory. As these piece ID paths are deleted, we will read the signed delete message files that they point to and decrement the first byte. Once this value reaches zero, none of the deleted pieces in the message can be reverted by a DB restoration; the proof is no longer necessary, and the file can be deleted.
```

Comment:
> Due to the aforementioned changes to how we run garbage collection, we can be sure that any piece which does not exist in the filter will not be reverted.

I feel like this could be reworded for more clarity. Firstly, I think we should restate the "aforementioned changes", i.e. that a satellite will never ask a storage node to delete a piece via garbage collection if there is any chance the satellite could restore that piece by restoring from a DB backup. Then, I would put "we can be sure that any piece which does not exist in the filter will not be reverted" in a separate sentence and try to get rid of the double negative, which I have a hard time parsing. Maybe something like "Because of this (the aforementioned changes), if a piece does not exist in the garbage collection bloom filter, it is safe to remove any proof of its deletion."
Not sure if that makes it easier to understand or not. I'll let you decide.